fanficfare/adapters/adapter_novelfull.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import absolute_import
+import logging
+import re
+
+# py2 vs py3 transition
+from ..six.moves.urllib import parse as urlparse
+
+from .base_adapter import BaseSiteAdapter
+
+
+logger = logging.getLogger(__name__)
+
+
+def getClass():
+    return NovelFullSiteAdapter
+
+
+class NovelFullSiteAdapter(BaseSiteAdapter):
+    @staticmethod
+    def getSiteDomain():
+        return "novelfull.com"
+
+    def getSiteURLPattern(self):
+        return r"https?://%s/(?P<name>.+).html?" % re.escape(self.getSiteDomain())
+
+    def extractChapterUrlsAndMetadata(self):
+        logger.debug("URL: %s", self.url)
+
+        data = self.get_request(self.url)
+
+        soup = self.make_soup(data)
+
+        self.story.setMetadata("title", soup.select_one("h3.title").text)
+        self.story.setMetadata(
+            "author",
+            " ".join(
+                s.text for s in soup.find("h3", text="Author:").fetchNextSiblings()
+            ),
+        )
+        self.story.setMetadata("authorId", self.story.getMetadata("author"))
+
+        status = soup.find("a", href=re.compile("status")).text
```

Comment:
Genre can be gotten very similarly, but use self.story.addToList().

Reply:
Just added